client/pd_service_discovery.go

```diff
@@ -378,8 +379,15 @@ func (c *pdServiceDiscovery) updateServiceMode() {
 	leaderAddr := c.getLeaderAddr()
 	if len(leaderAddr) > 0 {
 		clusterInfo, err := c.getClusterInfo(c.ctx, leaderAddr, c.option.timeout)
+		// If the method is not supported, we set it to pd mode.
 		if err != nil {
-			log.Warn("[pd] failed to get cluster info for the leader", zap.String("leader-addr", leaderAddr), errs.ZapError(err))
+			// TODO: it's a hack way to solve the compatibility issue.
```

Comment:
Probably we need to add service mode to the every rpc response from the pd/api and handle it gracefully.